AdGoBye/Indexer.cs

```diff
@@ -328,25 +339,35 @@ public static void PatchContent(Content content)
             }
         }
 
-        // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract - False positive
         if (Blocklist.Blocks is null) return;
         if (pluginOverridesBlocklist) return;
         if (content.VersionMeta.PatchedBy.Contains("Blocklist")) return;
         foreach (var block in Blocklist.Blocks.Where(block => block.Key.Equals(content.Id)))
         {
             try
             {
-                var unmatchedObjects = Blocklist.Patch(content.VersionMeta.Path + "/__data", block.Value.ToArray());
+                var unmatchedObjects = Blocklist.Patch(container, block.Value.ToArray());
                 if (Settings.Options.SendUnmatchedObjectsToDevs && unmatchedObjects is not null)
                     Blocklist.SendUnpatchedObjects(content, unmatchedObjects);
-                if (!Settings.Options.DryRun) content.VersionMeta.PatchedBy.Add("Blocklist");
             }
             catch (Exception e)
             {
                 Logger.Error(e, "Failed to patch {ID} ({path})", content.Id, content.VersionMeta.Path);
             }
         }
 
+        container.Bundle.file.BlockAndDirInfo.DirectoryInfos[1].SetNewData(container.AssetsFile.file);
+        using var writer = new AssetsFileWriter(file + ".clean");
+        container.Bundle.file.Write(writer);
+        // Moving the file without closing our access fails on NT.
+        writer.Close();
+        container.Bundle.file.Close();
+        container.Bundle.file.Close();
+
+        // TODO: Provide option to disable backup file?
```

Comment:
TODO present, something you want to finish here or save for later?

Reply:
I'll do this another time probably but I'm considering adding config fields for disabling the backup file and readding compression.